janusgraph-hadoop-parent/janusgraph-hadoop-core/src/main/java/org/janusgraph/hadoop/formats/cassandra/CassandraBinaryInputFormat.java

```diff
@@ -65,10 +65,14 @@
     @Override
     public RecordReader<StaticBuffer, Iterable<Entry>> createRecordReader(final InputSplit inputSplit, final TaskAttemptContext taskAttemptContext)
             throws IOException, InterruptedException {
-        columnFamilyRecordReader =
-                (ColumnFamilyRecordReader)columnFamilyInputFormat.createRecordReader(inputSplit, taskAttemptContext);
-        janusgraphRecordReader =
+        if (Boolean.getBoolean("use.cassandra.3")) { // see issue 172
+            janusgraphRecordReader = new CqlBridgeRecordReader();
+        } else { // the default case, Cassandra 2.1.9
+            columnFamilyRecordReader =
+                (ColumnFamilyRecordReader) columnFamilyInputFormat.createRecordReader(inputSplit, taskAttemptContext);
```

Comment:
Does it compile without the cast? if it compiles, lets get rid of it. otherwise we are OK here.